lib/media/time_ranges_utils.js

```diff
@@ -155,4 +155,49 @@ shaka.media.TimeRangesUtils = class {
     }
     return ret;
   }
+
+  /**
+   * This operation can be potentially EXPENSIVE and should only be done in
+   * debug builds for debugging purposes.
+   *
+   * @param {TimeRanges} oldRanges
+   * @param {TimeRanges} newRanges
+   * @return {?shaka.extern.BufferedRange}
```

Comment:
Actually, wait, one last thing.
This should probably specify that this is the last added range, chronologically.